homeassistant/components/overkiz/cover_entities/generic_cover.py

```diff
@@ -0,0 +1,211 @@
+        self, cancel_commands: list[OverkizCommand], stop_commands: list[OverkizCommand]
+    ) -> None:
+        """Cancel running execution or send stop command."""
+        success = await self.executor.async_cancel_command(cancel_commands)
+
+        if success:
+            return
+
+        if command := self.executor.select_command(*stop_commands):
+            # Fallback to available stop commands when no executions are found
+            # Stop commands don't work with all devices, due to a bug in Somfy service
+            await self.executor.async_execute_command(command)
+
+    @property
+    def is_opening(self) -> bool | None:
+        """Return if the cover is opening or not."""
+
+        if self.assumed_state:
+            return None
+
+        # Check if cover movement execution is currently running
+        if any(
+            execution.get("device_url") == self.device.device_url
+            and execution.get("command_name") in COMMANDS_OPEN + COMMANDS_OPEN_TILT
+            for execution in self.coordinator.executions.values()
+        ):
+            return True
+
+        # Check if cover is moving based on current state
+        is_moving = self.device.states.get(OverkizState.CORE_MOVING)
+        current_closure = self.device.states.get(OverkizState.CORE_CLOSURE)
+        target_closure = self.device.states.get(OverkizState.CORE_TARGET_CLOSURE)
+
+        if not is_moving or not current_closure or not target_closure:
+            return None
+
+        return cast(int, current_closure.value) > cast(int, target_closure.value)
+
+    @property
+    def is_closing(self) -> bool | None:
+        """Return if the cover is closing or not."""
+
+        if self.assumed_state:
+            return None
+
+        # Check if cover movement execution is currently running
+        if any(
+            execution.get("device_url") == self.device.device_url
+            and execution.get("command_name") in COMMANDS_CLOSE + COMMANDS_CLOSE_TILT
+            for execution in self.coordinator.executions.values()
+        ):
+            return True
+
+        # Check if cover is moving based on current state
+        is_moving = self.device.states.get(OverkizState.CORE_MOVING)
+        current_closure = self.device.states.get(OverkizState.CORE_CLOSURE)
+        target_closure = self.device.states.get(OverkizState.CORE_TARGET_CLOSURE)
+
+        if not is_moving or not current_closure or not target_closure:
+            return None
+
+        return cast(int, current_closure.value) < cast(int, target_closure.value)
+
+    @property
+    def extra_state_attributes(self) -> Mapping[str, Any] | None:
+        """Return the device state attributes."""
+        attr: Mapping[str, Any] = super().extra_state_attributes or {}
+
+        # Obstruction Detected attribute is used by HomeKit
+        if self.executor.has_state(OverkizState.IO_PRIORITY_LOCK_LEVEL):
+            attr[ATTR_OBSTRUCTION_DETECTED] = True
```

Comment:
@epenet any idea what is wrong here? This is the last remaining error that I cannot seem to fix.

```homeassistant/components/overkiz/cover_entities/generic_cover.py:190: error: Unsupported target for indexed assignment ("Mapping[str, Any]")  [index]```